spec/support/api_spec_helper.rb

```diff
@@ -16,19 +16,8 @@ module ApiSpecHelper
     "Accept"       => "application/json"
   }
 
-  API_STATUS = Rack::Utils::HTTP_STATUS_CODES.merge(0 => "Network Connection Error")
-
-  def parse_response
-    @code    = response.status
-    @result  = case @code
-               when Rack::Utils.status_code(:no_content)   then {}
-               when Rack::Utils.status_code(:unauthorized) then response.body
-               else JSON.parse(response.body)
-               end
-    @success = response.success?
-    @status  = API_STATUS[@code] || (@success ? Rack::Utils.status_code(:ok) : Rack::Utils.status_code(:bad_request))
-    @message = @result.kind_of?(Hash) ? @result.fetch_path("error", "message").to_s : @result
-    @success
+  def response_hash
+    JSON.parse(response.body)
```

Comment:
> There are several specs in authorization for which memoizing would have an undesirable effect, because they make multiple requests in the same test, and make expectations against the response after each.

Ahhh true. I don't usually write specs myself this way (as that's testing two behaviors - they should just request twice before they assert something on the response), but this is a very valid point and an odd thing for people to think about. :+1: